test/e2e/cleanup/submariner.go

```diff
@@ -0,0 +1,264 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleanup
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"sync"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/admiral/pkg/watcher"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	operatorv1alpha1 "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/controllers/uninstall"
+	operatorclient "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	operatorv1alpha1client "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned/typed/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	submarinerclientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+var _ = Describe("Submariner cleanup", func() {
+	When("the Submariner resource is deleted", testSubmarinerCleanup)
+})
+
+func testSubmarinerCleanup() {
+	_ = framework.NewFramework("submariner-cleanup")
+
+	var (
+		submarinerInterface  operatorv1alpha1client.SubmarinerInterface
+		submariner           *operatorv1alpha1.Submariner
+		routeAgentPodMonitor *podMonitor
+		gatewayPodMonitor    *podMonitor
+		brokerRestConfig     *rest.Config
+		stopCh               chan struct{}
+	)
+
+	BeforeEach(func() {
+		stopCh = make(chan struct{})
+
+		operatorClient, err := operatorclient.NewForConfig(framework.RestConfigs[framework.ClusterA])
+		Expect(err).To(Succeed())
+
+		submarinerInterface = operatorClient.SubmarinerV1alpha1().Submariners(framework.TestContext.SubmarinerNamespace)
+
+		submariner, err = submarinerInterface.Get(context.TODO(), submarinercr.SubmarinerName, metav1.GetOptions{})
+		Expect(err).To(Succeed())
+
+		brokerRestConfig = getBrokerRestConfig(submariner.Spec.BrokerK8sRemoteNamespace)
+		Expect(brokerRestConfig).ToNot(BeNil(), "No broker located")
+
+		routeAgentPodMonitor = startPodMonitor(names.AppendUninstall(names.RouteAgentComponent), stopCh)
+		gatewayPodMonitor = startPodMonitor(names.AppendUninstall(names.GatewayComponent), stopCh)
+	})
+
+	AfterEach(func() {
+		if submariner != nil {
+			submariner.ObjectMeta = metav1.ObjectMeta{
+				Name:        submariner.Name,
+				Labels:      submariner.Labels,
+				Annotations: submariner.Annotations,
+			}
+
+			_, err := submarinerInterface.Create(context.TODO(), submariner, metav1.CreateOptions{})
+			if err != nil {
+				framework.Errorf("Error re-creating Submariner: %v", err)
+			}
+
+			By("Re-created Submariner resource")
+		}
+	})
+
+	It("should run DaemonSets/Deployments to uninstall components", func() {
```

Comment:
Both the pods are DaemonSets.

Reply:
The network plugin syncer is a `Deployment` although we're not checking that right now..